connector-runtime/connector-runtime-core/src/main/java/io/camunda/connector/runtime/core/outbound/AbstractOutboundConnectorContext.java

```diff
@@ -16,51 +16,91 @@
  */
 package io.camunda.connector.runtime.core.outbound;
 
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
 import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.fasterxml.jackson.databind.node.TextNode;
 import io.camunda.connector.api.error.ConnectorException;
 import io.camunda.connector.api.outbound.OutboundConnectorContext;
 import io.camunda.connector.api.secret.SecretProvider;
 import io.camunda.connector.api.validation.ValidationProvider;
 import io.camunda.connector.runtime.core.AbstractConnectorContext;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
 
 public abstract class AbstractOutboundConnectorContext extends AbstractConnectorContext
     implements OutboundConnectorContext {
 
   protected final ObjectMapper objectMapper;
 
-  private final String jsonVariables;
+  private final Object variables;
+  private JsonNode variablesWithReplacedSecrets = null;
+
   private String jsonWithSecrets = null;
```

Comment:
Shouldnt we also cache the object that contains the replaced secrets?

Reply:
We do (in `variablesWithReplacedSecrets`) - this string cache was not used any more, I removed it